File: Assets/PayPalShopSDK/Scripts/PHP/getpayment.php
<?php
$_post = (array) filter_input_array(INPUT_POST, FILTER_SANITIZE_SPECIAL_CHARS);
$_get = (array) filter_input_array(INPUT_GET, FILTER_SANITIZE_SPECIAL_CHARS);
if(!isset($_post["authorization"]) || !isset($_get['mode']) || !isset($_post["payID"])){
    echo "Error 403: Forbidden";
    exit;
}

$ch = curl_init();
if($_get['mode'] == 'live')
    $url = 'https://api.paypal.com/v1/payments/payment/' . $_post['payID'];
else
    $url = 'https://api.sandbox.paypal.com/v1/payments/payment/' . $_post['payID'];
curl_setopt($ch, CURLOPT_URL, $url);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);

$headers = [
    'Content-Type: application/json',
    'Authorization: Bearer ' . $_post['authorization'],
];

curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);

$server_output = curl_exec ($ch);

curl_close ($ch);

echo $server_output;

?>

File: Assets/PayPalShopSDK/Scripts/API_Calls/GetPayment.cs
﻿using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class GetPayment : MonoBehaviour
{

	public string payID, accessToken;
	[HideInInspector]
	public GetPaymentInfoResponse API_Success;
	[HideInInspector]
	public ErrorResponse API_Error;

	private void Start()
	{
		StartCoroutine(APICall());
	}

	private void Success(string response)
	{
		API_Success = JsonUtility.FromJson<GetPaymentInfoResponse>(response);
	}

	private void Error(string response, string error)
	{
		if (error != "")
			API_Error = JsonUtility.FromJson<ErrorResponse>(response);
		if (API_Error == null)
		{
			API_Error = new ErrorResponse();
			API_Error.message = error;
		}
	}

	private IEnumerator APICall()
	{
		Dictionary<string, string> APIheaders = new Dictionary<string, string>();
		APIheaders.Add("Content-Type", "application/json");
		APIheaders.Add("Authorization", "Bearer " + accessToken);
		string url = ShopSettings.inst.IsLive() ? "https://api.paypal.com/v1/payments/payment/" : "https://api.sandbox.paypal.com/v1/payments/payment/";
		string endpointURL = url + payID;
		WWW www = new WWW(endpointURL, null, APIheaders);
		yield return www;
		if (www.error == null || www.error == "")
		{
			Success(www.text);
		}
		else
		{
			if (www.error.Contains("Unable to complete SSL connection"))
			{
				StartCoroutine(WorkAround());
			}
			else
			{
				Debug.Log("Error connecting to " + url + " Error: " + www.error + " Details: " + www.text);
				Error(www.text, www.error);
			}
		}
	}
	//A workaround for a problem in some Unity versions, where Unity gives an error saying the api.paypal.com SSL certificate is invalid for some reason. 
	//What happens in this function is basically the same as the APICall function, but the data is send to my website instead and from there to api.paypal.com and back.
	//If you have your own website and you want to implement this workaround there, you can find the PHP code in Scripts > PHP.
	private IEnumerator WorkAround()
	{
		WWWForm form = new WWWForm();
		form.AddField("authorization", accessToken);
		form.AddField("payID", payID);
		WWW www = new WWW("https://api.toolwareassets.com/v1/paypal/getpayment" + (ShopSettings.inst.IsLive() ? "?mode=live" : "?mode=sandbox"), form);
		yield return www;
		if (!www.text.Contains("error"))
		{
			Success(www.text);
		}
		else
		{
			Debug.Log("Connection Error: " + www.text);
			Error(www.text, www.text);
		}
	}
}
